src/api/roleApi.ts

```diff
@@ -71,18 +71,83 @@ export const assignUserRole = async (
     const data = {
       institution,
       username,
-      roles: [{ rolename, type: 'Role' }],
+      roles: [roleElement],
       type: 'User',
     };
+
     const response = await Axios.post(url, data, { headers, cancelToken });
     if (response.status === StatusCode.OK) {
       return response.data;
+    } else {
+      return { error: i18n.t('feedback:error.add_role') };
     }
   } catch (error) {
     if (error.response.status === StatusCode.CONFLICT) {
-      return { info: i18n.t('feedback:info.user_already_exists') };
+      // Update existing user instead of creating a new one
+      return await getUserAndAddRole(username, institution, roleElement, cancelToken);
     } else if (!Axios.isCancel(error)) {
       return { error: i18n.t('feedback:error.add_role') };
     }
   }
 };
+
+const getUserAndAddRole = async (
+  username: string,
+  institution: string,
+  newRole: UserRole,
+  cancelToken?: CancelToken
+) => {
+  try {
+    const existingUser = await getInstitutionUser(username, cancelToken);
+    if (existingUser) {
+      if (existingUser.error) {
+        // Forward error from fetch user call
+        return existingUser;
+      } else if (existingUser.institution !== institution) {
+        // A user can only have roles in one institution
```

Comment:
hvordan går dette an? Støtter vi ikke flere institusjoner? Hva med app-admins?

Reply:
en bruker kan kun logge inn via en institusjon, og dermed kun ha roller i denne. Alle som har rollen `App-admin` vil kunne endre alle institusjoner uavhengig av hvilken institusjon man logger inn via. Så hvis en person skal være `Institution-admin` for to ulike institusjoner må man ha to ulike brukere, og logge inn med riktig institusjon.
Det er i hvert fall sånn det er modellert atm, modeller er nederst på siden her: https://portal.dev.nva.aws.unit.no/apis/ddpsk7vp6h/Prod